client/image_build.go

```diff
@@ -123,6 +123,12 @@ func (cli *Client) imageBuildOptionsToQuery(options types.ImageBuildOptions) (ur
 	}
 	query.Set("labels", string(labelsJSON))
 
+	storageOptJSON, err := json.Marshal(options.StorageOpt)
```

Comment:
May want to add serialization tests to prevent regressions. I know some other options have them.